Pod/Classes/LSClockState.m

```diff
@@ -98,7 +108,7 @@ - (void)addSampleWithOriginMicros:(SInt64)originMicros
         self.currentOffsetAge++;
 
         // Remember what we've seen.
-        [self _persistSamplesToUserDefaults:self.samples.copy];
```

Comment:
Hi @bcronin !

> dispatch_async callback is referencing self

Yup. And, because this isn't using using a weak reference (i.e. `__weak typeof(self) weakSelf = self;`) the block should retain `self`.

> but there are no guarantees around when the dispatch may occur

Correct

> and, by extension, no guarantees that self will still be a valid reference inside the callback

I don't think this is true in an ARC environment, since, as I mentioned in the comment earlier, the block will keep a reference to `self`. While this is often discouraged as it leads to retain cycles, in this case, this helps, since it ensures that `self` won't be deallocated until the block executes.

Of course, if ARC is explicitly disabled in this project, you'd be correct. I don't think that's the case, though - if so, we'd see things like `[foo retain]` and `[foo release]` everywhere.

(You misspelled my name / github handle, FYI - & no worries.)

Reply:
Thanks, for the clarification. Ah, knew there was something unusual around the strong reference, but had it backwards (retain cycle, not a potential deallocation). It _has_ been a while since I've written any Objective-C :)